### PR TITLE
feat(my-account): My Account Filter Order History.

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**

### CODE REVIEW

**Review by kibo-chandradeeptalaha — APPROVED** [review]

File: components/KiboMobileFacetContainer.vue [hunk 1]
```diff
@@ -12,7 +12,9 @@
         />
       </div>
       <KiboFilterTiles :applied-filters="appliedFilters" @removeSelectedFilter="removeFilter" />
-      <hr class="filter-hr" />
+      <hr
+        :class="`${className === 'time-filter' ? 'filter-hr filter-hr--time-filter' : 'filter-hr'}`"
```
Comment (TriptiVermaXP) [inline components/KiboMobileFacetContainer.vue]: instead passing string classname, we can handle it by boolean value as isTimeFilter to added addition class as "", and filter-hr could be applied directly in both conditions.
```
<hr
        :class="`${'filter-hr--time-filter' : isTimeFilter }`"
        class="filter-hr"
      />
```

File: components/KiboMobileFacetContainer.vue [hunk 2]
```diff
@@ -36,6 +38,10 @@ export default defineComponent({
       type: Array,
       default: () => [],
     },
+    className: {
+      type: String,
+      default: "",
+    },
```
Comment (TriptiVermaXP) [inline components/KiboMobileFacetContainer.vue:42]: We can set isTimeFilter: false by default, wherever required to set class , can pass true value.

**Review by TriptiVermaXP — CHANGES_REQUESTED** [review]
Please check the review comments.

File: components/KiboMobileFacetContainer.vue (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply components/KiboMobileFacetContainer.vue]: Resolved

File: components/KiboMobileFacetContainer.vue (hunk 2 above)
Comment (kibo-geetanshuchhabra) [reply components/KiboMobileFacetContainer.vue:42]: Resolved

**Review by TriptiVermaXP — APPROVED** [review]

File: composables/helpers/buildOrdersFilterInput.ts [hunk 3]
```diff
@@ -6,12 +7,29 @@ export const buildOrdersFilterInput = (params: {
   const variables = {
     filter: "",
     startIndex: 0,
-    pageSize: 0,
+    pageSize: 20,
+  }
+
+  const searchFilters = ref([])
```
Comment (kibo-kevinwatts) [inline composables/helpers/buildOrdersFilterInput.ts]: this should not be a ref

File: composables/helpers/buildOrdersFilterInput.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { ref } from "@vue/composition-api"
 export const buildOrdersFilterInput = (params: {
```
Comment (kibo-kevinwatts) [inline composables/helpers/buildOrdersFilterInput.ts]: we need unit tests around this  buildOrdersFilterInput  function

File: pages/my-account/order-history.vue [hunk 5]
```diff
@@ -92,24 +97,37 @@ export default defineComponent({
...
     const barTitle = ref(myAccountText)
     const title = ref(orderHistoryText)
-    const appliedFilters = ref([])
     const isOpenOrderList = ref(false)
     const isOpenOrderItem = ref(false)
     const selectedOrder = ref({})
-    const facetAllOptions = ref([]) // @TODO need to be fetch from API
+    const filters = ref([])
+    const currentYear = new Date().getFullYear()
+    const facetAllOptions = ref([
```
Comment (kibo-kevinwatts) [inline pages/my-account/order-history.vue:113]: I like how you were able to come up with these filter values in a similar way we do the search page

File: pages/my-account/order-history.vue [hunk 6]
```diff
@@ -144,10 +162,46 @@ export default defineComponent({
...
+    }
+
     useAsync(async () => {
-      await getOrders({ filters: "" })
+      const facetsFromURL = getFacetsFromURL()
+      facetsFromURL.filters.forEach((filter) => {
+        facetAllOptions.value.find((facet) => facet.filterValue === filter).isApplied = true
+      })
+
+      facetAllOptions.value.forEach((facet) => {
```
Comment (kibo-kevinwatts) [inline pages/my-account/order-history.vue]: it looks like you're duplicating the state in multiple spots.
Theres a "filter" ref, and you're mutating the "facetAllOptions" and getting pulling values from the url?
Setup one approach and stick with it

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: composables/helpers/buildOrdersFilterInput.ts [hunk 7]
```diff
@@ -6,12 +7,29 @@ export const buildOrdersFilterInput = (params: {
...
+    for (const filters of params.filters) {
+      const filter = filters.split("-")
+      if (filter[0] === "M") {
+        searchFilters.value.push(`createDate gt ${getPastDateTimestamp(parseInt(filter[1]))}`)
+      } else if (filter[0] === "Y") {
+        const startDate = new Date(`${filter[1]}-01-01`).toISOString()
+        const endDate = new Date(`${filter[1]}-12-31`).toISOString()
+        searchFilters.value.push(`createDate gt ${startDate} and createDate lt ${endDate}`)
+      }
+    }
```
Comment (kibo-sushant) [inline composables/helpers/buildOrdersFilterInput.ts:29]: As a best practice always convert string to lowercase and then compare.
ex. if (filter[0].toLowerCase() === "m")

Comment (kibo-geetanshuchhabra) [reply composables/helpers/buildOrdersFilterInput.ts:29]: Resolved

File: composables/helpers/buildOrdersFilterInput.ts (hunk 3 above)
Comment (kibo-geetanshuchhabra) [reply composables/helpers/buildOrdersFilterInput.ts]: Resolved

File: composables/helpers/buildOrdersFilterInput.ts (hunk 4 above)
Comment (kibo-geetanshuchhabra) [reply composables/helpers/buildOrdersFilterInput.ts]: Resolved

File: pages/my-account/order-history.vue (hunk 6 above)
Comment (kibo-geetanshuchhabra) [reply pages/my-account/order-history.vue]: Resolved

File: composables/helpers/buildOrdersFilterInput.ts (hunk 7 above)
Comment (kibo-geetanshuchhabra) [reply composables/helpers/buildOrdersFilterInput.ts:29]: Resolved

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

**Pull request merged**